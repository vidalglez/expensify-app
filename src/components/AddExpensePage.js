import React from 'react';
import ReactDOM from 'react-dom';

const AddExpensePage = () => (
  <div>
    This is from my add expense component
  </div>
);

export default AddExpensePage;