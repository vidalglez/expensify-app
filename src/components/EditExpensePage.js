import React from 'react';
import ReactDOM from 'react-dom';

const EditExpensePage = () => (
  <div>
    This is from my edit expense component
  </div>
);

export default EditExpensePage;