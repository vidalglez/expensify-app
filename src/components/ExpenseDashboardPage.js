import React from 'react';
import ReactDOM from 'react-dom';

const ExpenseDashboardPage = () => (
  <div>
    This is from my dashboard component
  </div>
);

export default ExpenseDashboardPage;