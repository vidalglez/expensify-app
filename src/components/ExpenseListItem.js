import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{description}</h3>
      <span className="list-item__sub-title">{numeral(amount / 100).format('$0,0.00')}</span>
    </div>
    <h3 className="list-item__data">{moment(createdAt).format('MMMM Do, YYYY')}</h3>
  </Link>
);

export default ExpenseListItem;
