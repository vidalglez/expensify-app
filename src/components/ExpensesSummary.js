import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => {
    const expenseWord = `expense${props.expenseCount === 1 ? '' : 's'}`;
    const expensesTotal = numeral(props.expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {props.expenseCount} {expenseWord} totalling {expensesTotal}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const selectedExpenes = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: selectedExpenes.length,
        expensesTotal: selectExpensesTotal(selectedExpenes)
    };
}

export default connect(mapStateToProps)(ExpensesSummary);