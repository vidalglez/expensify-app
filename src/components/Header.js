import React from 'react';
import ReactDOM from 'react-dom';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header_content">
        <Link className="header__title" to="/dashboard" >
          <h1>Expensify</h1>
        </Link>
        <button className="button button__link" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);