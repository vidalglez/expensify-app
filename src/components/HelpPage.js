import React from 'react';
import ReactDOM from 'react-dom';

const HelpPage = () => (
  <div>
    This is from my help page component
  </div>
);

export default HelpPage;