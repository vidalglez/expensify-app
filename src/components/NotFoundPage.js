import React from 'react';
import ReactDOM from 'react-dom';
import {Link} from  'react-router-dom';

const NotFoundPage = () => (
   <div>
    404!!! - <Link to="/">Go home</Link>
   </div>
);

export default NotFoundPage;