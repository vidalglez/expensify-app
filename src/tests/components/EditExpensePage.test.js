import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';
let wrapper, editExpense, startRemoveExpense, history;

beforeEach(() => {
    editExpense = jest.fn();
    startRemoveExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage 
        editExpense={editExpense} 
        startRemoveExpense={startRemoveExpense} 
        history={history}
        expense={expenses[0]}
    />);
});

test('Should render EditExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[0].id, expenses[0]);
});

test('Should handle startRemoveExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(startRemoveExpense).toHaveBeenLastCalledWith({
        id: expenses[0].id
    });
});