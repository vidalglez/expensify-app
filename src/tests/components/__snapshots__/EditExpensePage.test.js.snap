// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 4500,
        "createdAt": 0,
        "description": "Water bill",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
